comp/dogstatsd/packets/pool_test.go

```diff
@@ -39,48 +38,22 @@ func TestPoolTelemetry(t *testing.T) {
 	telemetryMock, ok := telemetryComponent.(telemetry.Mock)
 	assert.True(t, ok)
 
-	var poolMetric []*dto.Metric
-	var pollPutMetric []*dto.Metric
-	var pollGetMetric []*dto.Metric
-
-	metricsFamily, err := telemetryMock.GetRegistry().Gather()
+	poolMetrics, err := telemetryMock.GetGaugeMetric("dogstatsd", "packet_pool")
+	assert.Nil(t, err)
```

Comment:
You should use `require.NoError(t, err)` as `poolMetrics` is used later in this function.